src/StackInterface.php

```diff
@@ -43,5 +17,5 @@ public function withoutMiddleware(MiddlewareInterface $middleware);
      *
      * @return ResponseInterface
      */
-    public function process(RequestInterface $request);
+    public function dispatch(RequestInterface $request);
```

Comment:
Pipelines as middleware is something we're exploring [for CakePHP](https://github.com/cakephp/cakephp/issues/9089) as well. It has some nice benefits in that pipelines and individual middleware are easily composed together.